### PR TITLE
js-task-parse-list: add tests

### PR DESCRIPTION
**Pull request by brainwashers**

### CODE REVIEW

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('should render list', () => {
+    cy.get('ul');
+  });
+
+  it('should have 11 items', () => {
+    cy.get('ul > li')
+      .then(($ul) => {
+        expect($ul, '11 items').to.have.length(11);
+      });
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:16]: there has to be a shorter version to test list length

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  });
+
+  it('1st item should be Vivian Harrell', () => {
+    cy.get('ul > li')
+      .then(($ul) => {
+        expect($ul.eq(0), 'first item').to.contain('Vivian Harrell');
+      });
+  });
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:24]: repalce all test cases below with a test case verifying data is sorted.

**Review by roman-mate — CHANGES_REQUESTED** [review]

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+describe('List', () => {
+  beforeEach(() => {
+    cy.visit('index.html');
+  });
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:6]: https://docs.cypress.io/guides/references/best-practices.html#Setting-a-global-baseUrl

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  it('should have 11 items', () => {
+    cy.get('ul > li').should('have.length', 11);
+  });
+
+  it('should be sorted by salary DSC', () => {
+
+    cy.get('ul > li').then(($ul) => {
+     
+      const list = [...$ul].map((row) => row.dataset.salary);
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:21]:
```suggestion
      const list = $ul.map((row) => row.dataset.salary.replace('$', '').replace(',', ''));
```

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        sortedList.push(listCopy[i].replace('$', '').replace(',', ''))
+      }
+
+      sortedList.sort((a,b) => b - a);
+
+      for (let i = 0; i < arrCopy.length; i++) {
+        if (arrCopy[i] === sortedList[i]) {
+          counter += 1;
+        }
+      }
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:38]:
```
if (list.length < 2) {
return true;
}

let prev = list[0];
for (let i = 1; i < list.length; i++) {
  if (list[i]< prev) { return false;}
  prev = list[i];
}```

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+describe('List', () => {
```
Comment (iBrianWarner) [inline cypress/integration/jsTaskParseListDOM.spec.js:3]:
```suggestion
describe('Parse list app ', () => {
```

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      const list = [...$ul].map((row) => row.dataset.salary.replace('$', '').replace(',', ''));
+      let prev = list[0];
+
+      for (let i = 1; i < list.length; i++) {
+        if (list[i] < prev) { 
+          prev = list[i];
+        };
+      };
+
+      expect(prev).to.equal(list[list.length -2]);
```
Comment (iBrianWarner) [inline cypress/integration/jsTaskParseListDOM.spec.js:28]:
```suggestion
      expect(prev).to.equal(list[list.length - 2]);
```

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  it('should have 11 items', () => {
+    cy.get('ul > li').should('have.length', 11);
+  });
+
+  it('should be sorted by salary DSC', () => {
+
+    cy.get('ul > li').then(($ul) => {
+      const list = [...$ul].map((row) => row.dataset.salary.replace('$', '').replace(',', ''));
```
Comment (iBrianWarner) [inline cypress/integration/jsTaskParseListDOM.spec.js:19]:
```suggestion
      const list = [...$ul].map((row) =>
        row.dataset.salary.replace('$', '').replace(',', ''));
```

**Review by iBrianWarner — APPROVED** [review]

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+  it('should have 11 items', () => {
+    cy.get('ul > li').should('have.length', 11);
+  });
+
+  it('should be sorted by salary DSC', () => {
+
+    cy.get('ul > li').then(($ul) => {
+      const list = [...$ul].map((row) => row.dataset.salary
+        .replace('$', '').replace(',', ''));
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:20]: break line at higher level concepts.
```suggestion
      const list = [...$ul].map(
          (row) => row.dataset.salaryreplace('$', '').replace(',', ''));
```

File: cypress/integration/jsTaskParseListDOM.spec.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+      let prev = list[0];
+
+      for (let i = 1; i < list.length; i++) {
+        if (list[i] < prev) { 
+          prev = list[i];
+        };
+      };
+
+      expect(prev).to.equal(list[list.length - 2]);
```
Comment (roman-mate) [inline cypress/integration/jsTaskParseListDOM.spec.js:29]: this code will fail for input: 

[10, 20, 10, 1, 0]

**Review by roman-mate — CHANGES_REQUESTED** [review]

File: cypress/integration/jsTaskParseListDOM.spec.js (hunk 6 above)
Comment (roman-mate) [reply cypress/integration/jsTaskParseListDOM.spec.js:3]: @brainwashers this is not done?

**Review by roman-mate — APPROVED** [review]

File: cypress/integration/jsTaskParseListDOM.spec.js (hunk 6 above)
Comment (brainwashers) [reply cypress/integration/jsTaskParseListDOM.spec.js]: done

**Pull request merged**